module-system/module-implementations/sov-accounts/src/lib.rs

```diff
@@ -82,54 +83,141 @@ impl<C: Context> sov_modules_api::Module for Accounts<C> {
-impl<'a, C> Accounts<C>
-where
-    C: Context,
-    C::Address: arbitrary::Arbitrary<'a>,
-    C::PublicKey: arbitrary::Arbitrary<'a>,
-{
-    /// Creates an arbitrary set of accounts and stores it under `working_set`.
-    pub fn arbitrary_workset(
-        u: &mut arbitrary::Unstructured<'a>,
-        working_set: &mut WorkingSet<C>,
-    ) -> arbitrary::Result<Self> {
-        use sov_modules_api::Module;
+    impl<'a, C> Arbitrary<'a> for AccountConfig<C>
+    where
+        C: Context,
+        C::PublicKey: Arbitrary<'a>,
+    {
+        fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+            // TODO we might want a dedicated struct that will generate the private key counterpart so
+            // payloads can be signed and verified
+            Ok(Self {
+                pub_keys: u.arbitrary_iter()?.collect::<Result<_, _>>()?,
+            })
+        }
+    }
 
-        let config: AccountConfig<C> = u.arbitrary()?;
-        let accounts = Accounts::default();
+    impl<C> proptest::arbitrary::Arbitrary for AccountConfig<C>
+    where
+        C: Context,
+        C::PrivateKey: proptest::arbitrary::Arbitrary,
+    {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<Self>;
+
+        fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+            any::<Vec<C::PrivateKey>>()
+                .prop_map(|keys| AccountConfig {
+                    pub_keys: keys.into_iter().map(|k| k.pub_key()).collect(),
+                })
+                .boxed()
+        }
+    }
 
-        accounts
-            .genesis(&config, working_set)
-            .map_err(|_| arbitrary::Error::IncorrectFormat)?;
+    impl<'a, C> Accounts<C>
+    where
+        C: Context,
+        C::Address: Arbitrary<'a>,
+        C::PublicKey: Arbitrary<'a>,
+    {
+        /// Creates an arbitrary set of accounts and stores it under `working_set`.
+        pub fn arbitrary_workset(
+            u: &mut Unstructured<'a>,
+            working_set: &mut WorkingSet<C>,
+        ) -> arbitrary::Result<Self> {
+            let config: AccountConfig<C> = u.arbitrary()?;
+            let accounts = Accounts::default();
+
+            accounts
+                .genesis(&config, working_set)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?;
+
+            Ok(accounts)
+        }
+    }
 
-        Ok(accounts)
+    impl<C> Accounts<C>
+    where
+        C: Context,
+        C::Address: proptest::arbitrary::Arbitrary,
+        C::PrivateKey: proptest::arbitrary::Arbitrary,
+    {
+        /// Creates an arbitrary set of accounts and stores it under `working_set`.
+        ///
+        /// We take the `WorkingSet` as `Arc<Mutex<_>>` so the strategy can freely implement
+        /// parallelism while preserving interior mutability safety. The return is a reference to a
+        /// promise of a strategy; hence, the `WorkingSet` lifetime must be locked to its lifetime.
+        pub fn arbitrary_proptest_workset(
```

Comment:
Where is this function called?

Reply:
This is equivalent (but for proptest) to the example here:

https://github.com/Sovereign-Labs/sovereign-sdk/blob/1c3f77f812e7f0e36e96db5066baa20fa225752f/fuzz/fuzz_targets/accounts_call_random.rs#L11-L25

It is a convenience function to allow us to fuzzy things faster, without worrying about the internals of the struct

Reply:
Since it's currently not used, the last commit removes it. If we need it in the future, we can just add